modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDynamicParameterTest.java

```diff
@@ -59,19 +64,28 @@ public void dropTables() {
         sql("DROP TABLE IF EXISTS t1");
     }
 
-    @ParameterizedTest
-    @EnumSource(value = ColumnType.class,
-            //    https://issues.apache.org/jira/browse/IGNITE-18789
-            //    https://issues.apache.org/jira/browse/IGNITE-18414
-            //    https://issues.apache.org/jira/browse/IGNITE-18345
-            names = {"DECIMAL", "NUMBER", "BITMASK", "DURATION", "DATETIME", "TIMESTAMP", "DATE", "TIME", "PERIOD"},
-            mode = Mode.EXCLUDE
-    )
-    void testMetadataTypesForDynamicParameters(ColumnType type) {
-        Object param = generateValueByType(RND.nextInt(), type);
-
-        assertQuery("SELECT typeof(?)").withParams(param).returns(toSqlType(type)).check();
-        assertQuery("SELECT ?").withParams(param).returns(param).columnMetadata(new MetadataMatcher().type(type)).check();
+    @Test
+    void testMetadataTypesForDynamicParameters() {
+        //    https://issues.apache.org/jira/browse/IGNITE-18789
+        //    https://issues.apache.org/jira/browse/IGNITE-18414
+        String[] excludedTypes = {"DECIMAL", "NUMBER", "BITMASK", "DURATION", "PERIOD"};
```

Comment:
Why is not the previous approach with @EnumSource ?

Reply:
cause it start-stop ignite instances for each type and consume huge time.

Reply:
you just removed the symptoms while the problem still exists

Reply:
ok, i revert it